Yoast/YoastSEO.js
# 1743 upgrade eslint config

## Summary

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* Upgrade `eslint-config-yoast` to `5.0.1`
* Applied ESLint auto fixes and easy to fix things like too long sentences.

The idea is to not create more ESLint problems in new code.
* Overwritten the local configuration to move the errors back to warnings.
* Bumped the allowed warnings up to make room for all the warnings.
* Creating 3 separate issues to tackle these error -> warning overrides.

## Test instructions

This PR can be tested by following these steps:

* Check with `grunt check` to see the ESLint problems. These are now a lot more.
* Test with `yarn test` if the specs are still working.
* Check manually if the results look like they are working still as a lot of code was touched. Though it is not very likely that things got broken.

Fixes #1743
